crates/holochain/src/conductor/handle.rs

```diff
@@ -423,6 +437,28 @@ impl<DS: DnaStore + 'static> ConductorHandleT for ConductorHandleImpl<DS> {
                     .map_err(holochain_p2p::HolochainP2pError::other);
                 respond.respond(Ok(async move { res }.boxed().into()));
             }
+            QueryGossipAgents {
+                since_ms,
+                until_ms,
+                arc_set,
+                respond,
+                ..
+            } => {
+                use holochain_sqlite::db::AsP2pAgentStoreConExt;
+                let env = { self.conductor.read().await.p2p_env(space) };
+                let res = env
+                    .conn()?
+                    .p2p_gossip_query_agents(since_ms, until_ms, (*arc_set).clone())
+                    // FIXME: This sucks we have to iterate through the whole vec just to add Arcs.
+                    // Are arcs really saving us that much?
```

Comment:
Ha, this is a case where I thought you were talking about DhtArcs here, not sync::Arcs, and was trying to wrap my head around it.

Well, the primitive type that kitsune deals with is basically Arc<KitsuneAgent>. So perhaps `p2p_gossip_query_agents` should be wrapping them in Arcs before returning. Is there any other reason not to? Other than the slight overhead of Arcs?

Reply:
lol, I take it you saw my comment.

Reply:
Yeah, the distinction seems clear to me in code, but then when we talk about it more colloquially and in comments, there is this unfortunate name collision

Reply:
Arc are a pain when you don't need them and you want an owned or mut value because you either need to clone the underlying value or try_unwrap it which is hard to get right so I'd prefer that the db didn't return arcs but I'm not going to object to it if everyone thinks it's the way to go.

Reply:
I have some comments in the code thinking the same thing, especially for op hashes, I think it's wasteful to put them all in Arcs especially when there could be thousands of them.